modules/shape/src/tps_trans.cpp

```diff
@@ -161,8 +161,9 @@ void ThinPlateSplineShapeTransformerImpl::warpImage(InputArray transformingImage
         for (int col = 0; col < theinput.cols; col++)
         {
             Point2f pt = _applyTransformation(shapeReference, Point2f(float(col), float(row)), tpsParameters);
-            mapX.at<float>(row, col) = pt.x;
-            mapY.at<float>(row, col) = pt.y;
+
+            mapX.at<float>(row, col) = col - (pt.x - col);
```

Comment:
This estimation is wrong. According to math evaluations you provide in description here should be
`mapX(x1, y1) = x1 - deltaX` where `deltaX = x1 - x`. You could consider `x1` as new `x`, but in this case you should somehow rename old `x` to retain it in the formula. Therefore if you substitute `deltaX` it would be `mapX(x1, y1) = x1 - (x1 -x)`. So after renaming you get `mapX(x, y) = x - (x - oldX)` instead of `mapX(x, y) = x - (x1 - x)`
For example the following set of points will be evaluated by the current code incorrectly:
src(0,0)->dst(1,0)
src(1,0)->dst(1,1)
src(1,1)->dst(0,1)
src(0,1)->dst(0,0)
In accordance with the formula dst(0,0) will be back projected to src(-pt.x, -pt.y)=src(-1, 0)(in accordance with `_applyTransformation` behaviour) but it should be src(0,1)